src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -63,68 +65,84 @@ public CommandManagerWindow(DriverCommandManager commandManager) {
-		c.weighty = 1;
+		c.weighty = 0;
 		content.add(observerListField, c);
 		updateObserverListField();
 
 		currentCommandField = new JTextArea("");
 		currentCommandField.setEditable(false);
-		c.fill = GridBagConstraints.BOTH;
-		c.weightx = 1;
-		c.gridx = 0;
 		c.weighty = 1;
 		content.add(currentCommandField, c);
 		updateCurrentCommandField();
 
 		parser = new JsonParser();
 
 		Container container = getImportContainer();
-		c.fill = GridBagConstraints.BOTH;
-		c.weightx = 1;
-		c.gridx = 0;
-		c.weighty = 1;
-		content.add(container,c);
+		c.weighty = 2;
+		content.add(container, c);
 
 		JButton btnExportCommand = new JButton("Export commands ");
-		btnExportCommand.addActionListener((ActionEvent e) ->  this.exportCommands());
-		c.fill = GridBagConstraints.BOTH;
-		c.weightx = 1;
-		c.gridx = 0;
-		c.weighty = 1;
+		btnExportCommand.addActionListener((ActionEvent e) -> this.exportCommands());
+		c.weighty = 3;
 		content.add(btnExportCommand, c);
 
 		JButton btnClearCommand = new JButton("Clear command");
 		btnClearCommand.addActionListener((ActionEvent e) -> this.clearCommand());
-		c.fill = GridBagConstraints.BOTH;
-		c.weightx = 1;
-		c.gridx = 0;
-		c.weighty = 1;
+		c.weighty = 4;
 		content.add(btnClearCommand, c);
 
 		JButton btnClearObservers = new JButton("Delete observers");
 		btnClearObservers.addActionListener((ActionEvent e) -> this.deleteObservers(btnClearObservers));
-		c.fill = GridBagConstraints.BOTH;
-		c.weightx = 1;
-		c.gridx = 0;
-		c.weighty = 1;
+		c.weighty = 5;
 		content.add(btnClearObservers, c);
 
 		JButton btnRunCommand = new JButton("Run command");
 		btnRunCommand.addActionListener((ActionEvent e) -> this.runCommand());
-		c.fill = GridBagConstraints.BOTH;
-		c.weightx = 1;
-		c.gridx = 0;
-		c.weighty = 1;
+		c.weighty = 6;
 		content.add(btnRunCommand, c);
 
+		CommandFactory.getInstance().addObserver(this);
 		DefaultListModel model = new DefaultListModel();
 		CommandHistoryController.setListModel(model);
 		JList commandHistoryList = new JList(model);
 		commandHistoryList.setSelectionMode(ListSelectionModel.SINGLE_INTERVAL_SELECTION);
-		commandHistoryList.addListSelectionListener(new SelectHistoryListOptionListener(commandManager, commandHistoryList));
+		commandHistoryList
+				.addListSelectionListener(new SelectHistoryListOptionListener(commandManager, commandHistoryList));
+		c.weighty = 6;
+		content.add(new JScrollPane(commandHistoryList), c);
 
+		factoryModel = new DefaultListModel();
+		JList commandFactoryList = new JList(factoryModel);
+		JButton btnRunCommandFromFactory = new JButton("Run command");
+		JButton btnRemoveCommandFromFactory = new JButton("Remove command");
+		btnRunCommandFromFactory.addActionListener((ActionEvent event) -> {
+			if (commandFactoryList.getSelectedIndex() != -1) {
+				try {
+					CommandFactory.getInstance().runCommand((String) commandFactoryList.getSelectedValue());
+				} catch (CloneNotSupportedException e1) {
+					e1.printStackTrace();
+				}
```

Comment:
It is hard to understand how:
"CommandFactory.getInstance().runCommand((String) commandFactoryList.getSelectedValue())" could cause CloneNotSupportedException.
Exception should be avoided or changed to a different (maybe new) one.